### PR TITLE
Plugins: Fix manifest validation

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
This PR fixes validation of the checksums in plugin manifests. I realized on working with plugin manifest that the original logic was comparing string representations of checksum bytes to hex encodings, which doesn't quite work.

I fixed an existing test to catch this issue, and added a test for the case where a plugin is missing files listed in the manifest.

### CODE REVIEW

File: pkg/plugins/manifest.go [hunk 1]
```diff
@@ -80,8 +82,9 @@ func readPluginManifest(body []byte) (*pluginManifest, error) {
 	return manifest, nil
 }
 
-// GetPluginSignatureState returns the signature state for a plugin
-func GetPluginSignatureState(plugin *PluginBase) PluginSignature {
+// getPluginSignatureState returns the signature state for a plugin.
+func getPluginSignatureState(log log.Logger, plugin *PluginBase) PluginSignature {
+	log.Debug("Getting signature state of plugin...", "plugin", plugin.Id)
```
Comment (bergquist) [inline pkg/plugins/manifest.go:87]: why `...` ? :)

File: pkg/plugins/manifest.go [hunk 2]
```diff
@@ -100,20 +103,24 @@ func GetPluginSignatureState(plugin *PluginBase) PluginSignature {
...
 		}
 		defer f.Close()
 
 		h := sha256.New()
 		if _, err := io.Copy(h, f); err != nil {
+			log.Warn("Couldn't read plugin file", "plugin", plugin.Id, "filename", fp)
 			return PluginSignatureModified
 		}
-		sum := string(h.Sum(nil))
+		sum := hex.EncodeToString(h.Sum(nil))
```
Comment (bergquist) [inline pkg/plugins/manifest.go:121]: How did this work before? :/

**Review by bergquist — APPROVED** [review]
LGTM

Great that you added test manifest files. :+1:

File: pkg/plugins/manifest.go (hunk 2 above)
Comment (aknuds1) [reply pkg/plugins/manifest.go:121]: @bergquist I think it was simply broken? When I updated a test's plugin to have a signed manifest, it got so far as to hit this bug.

File: pkg/plugins/manifest.go (hunk 1 above)
Comment (aknuds1) [reply pkg/plugins/manifest.go:87]: @bergquist I can remove it, I just tend to use it in case of lengthy operations.

**Pull request merged**